### PR TITLE
ISS-207 # Parameterized step - simple values and CSV values

### PR DESCRIPTION
**Pull request by nirmalchandra**

### CODE REVIEW

File: core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+public class ZeroCodeParameterizedProcessorImpl implements ZeroCodeParameterizedProcessor {
+    private static final Logger LOGGER = getLogger(ZeroCodeParameterizedProcessorImpl.class);
+    public static final String VALUE_SOURCE_KEY = "0";
+
+    private final ObjectMapper objectMapper;
+
+    private final CsvParser csvParser;
+
+    @Inject
+    public ZeroCodeParameterizedProcessorImpl(ObjectMapper objectMapper, CsvParser csvParser, CsvParser csvParser1) {
```
Comment (officiallysameer) [inline core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java]: What is the use of two CsvParser parameters here

File: core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessor.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package org.jsmart.zerocode.core.engine.preprocessor;
+
+import org.jsmart.zerocode.core.domain.ScenarioSpec;
+import org.jsmart.zerocode.core.domain.Step;
+
+public interface ZeroCodeParameterizedProcessor {
+
+    @Deprecated
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessor.java]: `Deprecated` methods and implementations can be deleted now(than **later**) as this is not in use yet.

File: core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    @Override
+    public ScenarioSpec processParameterized(ScenarioSpec scenario, int iteration) {
+
+        if(scenario.getParameterized() == null){
+
+            return scenario;
+
+        } else if (scenario.getParameterized().getValueSource() != null) {
+
+            return resolveParamsValues(scenario, iteration);
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java]: tests for this?

File: core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+        } else if (scenario.getParameterized().getCsvSource() != null) {
+
+            return resolveParamsCsv(scenario, iteration);
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeParameterizedProcessorImpl.java]: tests?

**Comment by authorjapps** [comment]
@officiallysameer , can u approve this now?

**Comment by authorjapps** [comment]
solves #283

**Pull request merged**